dissect/cstruct/expression.py

```diff
@@ -1,84 +1,284 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Dict
+from typing import TYPE_CHECKING, Callable, Dict
+
+from dissect.cstruct.exceptions import ExpressionParserError
 
 if TYPE_CHECKING:
     from dissect.cstruct import cstruct
 
 
+class ExpressionTokenizer:
+    def __init__(self, expression: str):
+        self.expression = expression
+        self.i: int = 0
+        self.tokens = []
+
+    def equal(self, token: str, expected: str) -> bool:
+        if isinstance(expected, set):
+            return token in expected
+        else:
+            return token == expected
+
+    def alnum(self, token: str) -> bool:
+        return token.isalnum()
+
+    def alpha(self, token: str) -> bool:
+        return token.isalpha()
+
+    def digit(self, token: str) -> bool:
+        return token.isdigit()
+
+    def digitHex(self, token: str) -> bool:
+        return token.isdigit() or token in {"a", "b", "c", "d", "e", "f", "A", "B", "C", "D", "E", "F"}
```

Comment:
```suggestion
    def hexdigit(self, token: str) -> bool:
        return token in string.hexdigits
```
And add a `import string` at the top.